### ISSUE TITLE
what if we use ec2's instead of fargate and EFS?

### ISSUE THREAD

**Issue by t0mZ4** [BrianHaak/cloud-plot#2]
Hi
I like your initiative here, but as you say, it is probably not the most cost-effective approach.

What if we deploy the docker to EC2s and use SSDs instead? We can use terraform to automate the process of EC2 spins.

these guys did it fairly inexpensive I would say:

https://www.reddit.com/r/chia/comments/mbm86u/plotting_in_the_cloud/

"Plotting 24 TBs in 12.5 hours using AWS: $730 (this includes the costs of the virtual servers and the virtual SSDs)."

and he says in retrospect he would do it with C5.xlarge

**Comment by BrianHaak** [comment]
Yes, I believe EC2 with EBS SSDs would be cheaper than EFS. Initially I did not realize that EFS charged for data over the network since it mounts as NFS. ECS alone can only reach 200 GB storage without resorting to EFS so I think it is out for now.
It would take some testing to find the best instance type.

**Comment by t0mZ4** [comment]
I have some aws credits i am willing to spare to compare instance types or to test the code.
do you want to make a modified branch of this project to work with EC2 instead?

I can do it manually I guess, but you seem to have the devOP skills to make it scripted/automated.

Also it would be nice to be able to set # of instances to spawn at once.

what do you think?

**Comment by renanwilliam** [comment]
I think that we can optimize it using 7zip + s3 to reduce the final bundle size and transfer out charges. Makes sense?

**Comment by siscape** [comment]
Hi folks -- I work at AWS and would be happy to provide a some resources to prove this out. As of now in my own testing NVME spot instances seem to be the best. I like how slick the orchestration is on the Fargate though. Also some sort of log aggregation on ELK, Prometheus or Cloudwatch would be awesome. Maybe we can start a little dev team on this. Ping me if anyone is interested.
